web/webkit/src/main/scala/net/liftweb/http/ContentParser.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2007-2015 WorldWide Conferencing, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.liftweb
+package http
+
+import java.io.InputStream
+
+import net.liftweb.common.Box
+import net.liftweb.util.Helpers
+
+import scala.xml.NodeSeq
+
+/** Objects which can parse content should implement this trait. See LiftRules.contentParsers */
+trait ContentParser {
+  /** The template filename suffixes this parser is for, e.g. "html", "md", "adoc", etc. */
+  def templateSuffixes: Seq[String]
+
+  /** Given an InputStream to a resource with a matching suffix, this function should provide the corresponding
+    * NodeSeq ready to serve
+    * @return
+    */
+  def parse(content:InputStream): Box[NodeSeq]
+
+  /** Called to auto-surround the content when needed, i.e. when this content is a top-level template */
+  def surround(content:NodeSeq): NodeSeq
+}
+
+object ContentParser {
+  /** Convenience function to convert a simple String => Box[NodeSeq] content parser function into a
+    * InputStream => Box[NodeSeq] function for satisfying the ContentParser contract.
```

Comment:
To continue on the scaladoc vein, I've been wrapping types and other code-related stuff in Scaladoc in backticks so they're styled appropriately. See #1650 for more—definitely willing to discuss, mind you. This just makes things easier to read in the generated documentation, IMO, while not interfering too much with reading the docs in code.